<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindowClass</class>
 <widget class="QMainWindow" name="MainWindowClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1134</width>
    <height>902</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout_28">
    <item>
     <widget class="QTabWidget" name="tab_3">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Corner Detection</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <property name="leftMargin">
           <number>10</number>
          </property>
          <property name="topMargin">
           <number>10</number>
          </property>
          <property name="rightMargin">
           <number>10</number>
          </property>
          <item>
           <widget class="QPushButton" name="harrisBrowse_2">
            <property name="text">
             <string>Browse</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <widget class="QLabel" name="label">
              <property name="text">
               <string>Threshold</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="lineEdit"/>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_29">
            <item>
             <widget class="QPushButton" name="apply_lambda_2">
              <property name="text">
               <string>Apply lambda -</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="apply_corner_det_2">
              <property name="text">
               <string>Apply Harris</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_4">
              <item>
               <widget class="QLabel" name="input_color_image_corner_detection_2">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="input_gray_image_corner_detection_2">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QLabel" name="corner_detected_output_2">
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>Template Matching</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <property name="spacing">
           <number>20</number>
          </property>
          <property name="leftMargin">
           <number>10</number>
          </property>
          <property name="topMargin">
           <number>10</number>
          </property>
          <property name="rightMargin">
           <number>10</number>
          </property>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_5" stretch="8,1,8">
            <item>
             <widget class="QLabel" name="match_image">
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="0,0,0,0">
              <property name="spacing">
               <number>3</number>
              </property>
              <property name="sizeConstraint">
               <enum>QLayout::SetDefaultConstraint</enum>
              </property>
              <property name="topMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QPushButton" name="match_browse">
                <property name="text">
                 <string>Browse Image</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="temp_browse">
                <property name="text">
                 <string>Browse Temp</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QComboBox" name="match_type">
                <property name="currentText">
                 <string>ssd</string>
                </property>
                <item>
                 <property name="text">
                  <string>ssd</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>ncc</string>
                 </property>
                </item>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="match_apply">
                <property name="text">
                 <string>Apply</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QLabel" name="temp_image">
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_12" stretch="5,0">
            <item>
             <widget class="QLabel" name="match_output">
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_27">
              <item>
               <widget class="QLabel" name="label_19">
                <property name="text">
                 <string>Elapsed Time</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="time_elapsed_2">
                <property name="text">
                 <string>time</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_4">
       <attribute name="title">
        <string>Sift</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_15">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_9" stretch="5,1,5,0">
          <property name="leftMargin">
           <number>10</number>
          </property>
          <property name="topMargin">
           <number>10</number>
          </property>
          <property name="rightMargin">
           <number>10</number>
          </property>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_7">
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_7">
              <item>
               <widget class="QPushButton" name="sift_img1_btn">
                <property name="text">
                 <string>Browse img1</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="sift_img1_label">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_6">
              <item>
               <widget class="QPushButton" name="sift_img2_btn">
                <property name="text">
                 <string>Browse img2</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="sift_img2_label">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_8">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_13">
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_8">
                <property name="spacing">
                 <number>10</number>
                </property>
                <item>
                 <widget class="QLabel" name="label_6">
                  <property name="text">
                   <string>Threshold</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="threshold_sift_input">
                  <property name="text">
                   <string>0.001</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_9">
                <property name="spacing">
                 <number>10</number>
                </property>
                <item>
                 <widget class="QLabel" name="label_7">
                  <property name="text">
                   <string>Level</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="level_sift_input">
                  <property name="text">
                   <string>4</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_12">
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_11">
                <property name="spacing">
                 <number>10</number>
                </property>
                <item>
                 <widget class="QLabel" name="label_9">
                  <property name="text">
                   <string>scale factor</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="scale_factor_sift_input">
                  <property name="text">
                   <string>0.5</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_10">
                <property name="spacing">
                 <number>10</number>
                </property>
                <item>
                 <widget class="QLabel" name="label_8">
                  <property name="text">
                   <string>sigma</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="sigma_sift_input">
                  <property name="text">
                   <string>1.5</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_14">
              <item>
               <widget class="QComboBox" name="sift_type">
                <item>
                 <property name="text">
                  <string>ssd</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>ncc</string>
                 </property>
                </item>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="apply_sift">
                <property name="text">
                 <string>Apply</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QLabel" name="output_sift_img">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_26">
            <item>
             <widget class="QLabel" name="time">
              <property name="text">
               <string>Elapsed Time</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="time_elapsed">
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_5">
       <attribute name="title">
        <string>Key Points</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_25">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_10" stretch="5,1,5">
          <property name="leftMargin">
           <number>10</number>
          </property>
          <property name="topMargin">
           <number>10</number>
          </property>
          <property name="rightMargin">
           <number>10</number>
          </property>
          <item>
           <widget class="QPushButton" name="in_key_btn">
            <property name="text">
             <string>Browse</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_13">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_17">
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_18">
                <property name="spacing">
                 <number>10</number>
                </property>
                <item>
                 <widget class="QLabel" name="label_13">
                  <property name="text">
                   <string>Threshold</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="th_l">
                  <property name="text">
                   <string>0.001</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_19">
                <property name="spacing">
                 <number>10</number>
                </property>
                <item>
                 <widget class="QLabel" name="label_14">
                  <property name="text">
                   <string>Level</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="lev_l">
                  <property name="text">
                   <string>4</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_20">
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_21">
                <property name="spacing">
                 <number>10</number>
                </property>
                <item>
                 <widget class="QLabel" name="label_15">
                  <property name="text">
                   <string>scale factor</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="sc_l">
                  <property name="text">
                   <string>0.5</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_22">
                <property name="spacing">
                 <number>10</number>
                </property>
                <item>
                 <widget class="QLabel" name="label_16">
                  <property name="text">
                   <string>sigma</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="su_l">
                  <property name="text">
                   <string>1.5</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_23">
              <item>
               <widget class="QPushButton" name="app_key_btn">
                <property name="text">
                 <string>Apply</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_24">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_16">
              <item>
               <layout class="QVBoxLayout" name="verticalLayout_11">
                <item>
                 <widget class="QLabel" name="in_key">
                  <property name="text">
                   <string/>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QLabel" name="out_key">
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1134</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
